### ISSUE TITLE
[Test Failure] Build error MSB4018 occur after updating package ‘Microsoft.NETCore.App’ in .NET Core Console templates

### ISSUE THREAD

**Issue by v-luzh** [NuGet/Home#4233]
**NuGet Version:** NuGet 4.0.0-RC3-2188
**VS Version:** D15PreRel 26108.00
**OS Version:** Windows 10 RS1_Release 14393.722.170106-1842

**Repro Steps:**
1.	New a Visual C#->.NET Core->Console App (.NET Core) template.
2.	Build the solution, no error.
3.	Right-click on the project and click ‘Manage NuGet Packages…’.
4.	Update package ‘Microsoft.NETCore.App’.
5.	Build the solution after updating and open Error List Window.

**Expected Result:**
No build error in Error List Window.

**Actual Result:**
Build error MSB4018 occur in Error List Window as below screenshot.
![builderrorconsolenetcore](https://cloud.githubusercontent.com/assets/20632625/21761382/872e702e-d68d-11e6-8b1a-c4690e2787e7.PNG)

**Note:**
1. It also repro in ‘Class Library (.NET Core)’ template.
2. It didn't repro on D15PreRel 26105.01+NuGet Client 4.0.0-RC3-2177.

**Error Log:**
Error	MSB4018	The "ResolvePackageDependencies" task failed unexpectedly.
System.IO.FileNotFoundException: Could not load file or assembly 'System.IO.FileSystem, Version=4.0.1.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a' or one of its dependencies. The system cannot find the file specified.
File name: 'System.IO.FileSystem, Version=4.0.1.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a'
   at Microsoft.NET.Build.Tasks.LockFileCache.LoadLockFile(String path)
   at Microsoft.NET.Build.Tasks.LockFileCache.GetLockFile(String path)
   at Microsoft.NET.Build.Tasks.ResolvePackageDependencies.get_LockFile()
   at Microsoft.NET.Build.Tasks.ResolvePackageDependencies.ReadProjectFileDependencies()
   at Microsoft.NET.Build.Tasks.ResolvePackageDependencies.ExecuteCore()
   at Microsoft.NET.Build.Tasks.TaskBase.Execute()
   at Microsoft.Build.BackEnd.TaskExecutionHost.Microsoft.Build.BackEnd.ITaskExecutionHost.Execute()
   at Microsoft.Build.BackEnd.TaskBuilder.<ExecuteInstantiatedTask>d__26.MoveNext()

WRN: Assembly binding logging is turned OFF.
To enable assembly bind failure logging, set the registry value [HKLM\Software\Microsoft\Fusion!EnableLog] (DWORD) to 1.
Note: There is some performance penalty associated with assembly bind failure logging.
To turn this feature off, remove the registry value [HKLM\Software\Microsoft\Fusion!EnableLog].
	ConsoleNetCore	C:\Program Files (x86)\Microsoft Visual Studio\2017\Enterprise\MSBuild\Sdks\Microsoft.NET.Sdk\build\Microsoft.PackageDependencyResolution.targets	154	
Error	MSB4018	The "GenerateDepsFile" task failed unexpectedly.
System.IO.FileNotFoundException: Could not load file or assembly 'System.IO.FileSystem, Version=4.0.1.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a' or one of its dependencies. The system cannot find the file specified.
File name: 'System.IO.FileSystem, Version=4.0.1.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a'
   at Microsoft.NET.Build.Tasks.GenerateDepsFile.ExecuteCore()
   at Microsoft.NET.Build.Tasks.TaskBase.Execute()
   at Microsoft.Build.BackEnd.TaskExecutionHost.Microsoft.Build.BackEnd.ITaskExecutionHost.Execute()
   at Microsoft.Build.BackEnd.TaskBuilder.<ExecuteInstantiatedTask>d__26.MoveNext()

WRN: Assembly binding logging is turned OFF.
To enable assembly bind failure logging, set the registry value [HKLM\Software\Microsoft\Fusion!EnableLog] (DWORD) to 1.
Note: There is some performance penalty associated with assembly bind failure logging.
To turn this feature off, remove the registry value [HKLM\Software\Microsoft\Fusion!EnableLog].
	ConsoleNetCore	C:\Program Files (x86)\Microsoft Visual Studio\2017\Enterprise\MSBuild\Sdks\Microsoft.NET.Sdk\build\Microsoft.NET.Sdk.targets	76

**Comment by v-nancyj** [comment]
Creating a Visual C#->.NET Core->Console App (.NET Core) template, install Newtonsoft.json package, build it, The issue (error MSB4018 ) also repro.

**Comment by jainaashish** [comment]
internal bug# 365633

**Comment by rrelyea** [comment]
This fix is going into msbuild and sdk by RC3 timeframe. So closing this issue.

**Comment by zhivchick** [comment]
I've got same error in VS 2017 RC3, after VS update. There is  "System.IO.FileSystem.dll" in MSBuild folder, but it doesn't seem to work. Are there any workarounds?
<img width="1280" alt="2017-01-31 0 39 20" src="https://cloud.githubusercontent.com/assets/5101810/22442930/185ef84a-e74e-11e6-8037-e241fabb202c.png">

**Comment by rrelyea** [comment]
@rainersigwald - What is the status of the msbuild + façade DLLs? This user saw with RC3?

**Comment by rainersigwald** [comment]
The NuGet tasks should ship the façades you need for RTW. Our bug is https://github.com/Microsoft/msbuild/issues/1542 and was too complex to deliver for RTW.

@jeffkl

**Comment by qbit86** [comment]
@zhivchick I've got the same issue after recent update. Any solution?

**Comment by docrinehart** [comment]
Encountered this issue after updating VS2017 to RC3_26127.0

**Comment by qbit86** [comment]
@zhivchick @docrinehart I've managed to solve this issue, but don't remember actual steps. I had to set `<EnableDefaultCompileItems>` to `false` since corresponding default behavior [has changed](https://github.com/dotnet/core/blob/master/release-notes/1.0/sdk/1.0-rc3-default-compile-items.md). Removed versioned `<PackageReference>`s to `Microsoft.NETCore.App` and `NETStandard.Library` since they are redundant now. I've created empty projects for `netstandard` library and `netcoreapp` executable to see how new default templates look like, and fixed existing projects accordingly.

See also: [Updating .NET Core csproj Projects](https://blogs.msdn.microsoft.com/dotnet/2017/01/30/announcing-net-core-net-native-and-nuget-updates-in-vs-2017-rc/#updating-net-core-projects)
